test/model/src/core/kv_table.cpp

```diff
@@ -282,14 +282,14 @@ kv_table::truncate(kv_transaction_ptr txn, const data_value &start, const data_v
         for (auto i = stop_iter; i != _data.end(); i++) {
             if (i->second.has_prepared())
                 throw known_issue_exception("WT-13232");
-            if (!i->second.exists() || i->second.implicit())
+            if (!i->second.exists(txn) || i->second.implicit())
                 continue;
             break;
         }
         for (auto i = std::reverse_iterator(start_iter); i != _data.rend(); i++) {
             if (i->second.has_prepared())
                 throw known_issue_exception("WT-13232");
-            if (!i->second.exists() || i->second.implicit())
+            if (!i->second.exists(txn) || i->second.implicit())
```

Comment:
I think this loop has an off by one bug. 
https://en.cppreference.com/w/cpp/container/map
map::lower_bound() is not less than the specified element, i.e >= the specified element. That includes the specified element.  Shouldn't this loop not include the specified element since is looking for elements before the range?

Reply:
The reverse iterator actually moves to the previous element. It feels counterintuitive, but if you think about it, that's exactly what you need to do so that reversing the `end` iterator gives you the last element: Calling `v.end()` on a collection creates an iterator that points past the end of the collection, so for `std::reverse_iterator(v.end())` to the last element of the collection, it must move the iterator.

Here is a small example:
```
#include <vector>
#include <iostream>
#include <iterator>

int main(void) {
	std::vector<int> v;
	for (int i = 0; i <= 10; i++)
		v.push_back(i);

	auto iter = v.begin() + 5;
	std::cout << *iter << std::endl; // Prints 5

	auto rev_iter = std::reverse_iterator(iter);
	std::cout << *rev_iter << std::endl; // Prints 4

	auto rev_iter_end = std::reverse_iterator(v.end());
	std::cout << *rev_iter_end << std::endl; // Prints 10
	return 0;
}
```